lib/colorizer.rb

```diff
@@ -1,13 +1,29 @@
 class Colorizer
-  def magenta(string)
-    "\e[35m#{string}\e[0m"
+  def initialize(color_count)
+    @color_count = color_count
   end
 
-  def cyan(string)
-    "\e[36m#{string}\e[0m"
+  def symbol_1(string)
+    colorize(string, 34)
   end
 
-  def yellow(string)
-    "\e[33m#{string}\e[0m"
+  def symbol_2(string)
+    colorize(string, 36)
+  end
+
+  def position(string)
+    colorize(string, 33)
+  end
+
+  private
+
+  attr_reader :color_count
+
+  def supported?
+    color_count >= 8
+  end
+
+  def colorize(string, code)
+    supported? ? "\e[#{code}m#{string}\e[0m" : string
```

Comment:
This is doesn't look very friendly as just a literal in the body of this method. How could we communicate what this does more effectively?